src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -329,23 +333,22 @@ void DockerVirtualMachinePrivate::probe(const QObject *context,
                 [=, runner = runner.get()](int exitCode, QProcess::ExitStatus exitStatus) {
             if (exitStatus != QProcess::NormalExit || exitCode != 0) {
                 if (context_)
-                    functor({}, false);
+                    functor(*state, false);
```

Comment:
The VM is considered existing when the image exists, no matter if the container exists. The fact that the state couldn't be fully determined is signalized by the bool flag.

Reply:
You are right. That's the way it should have always been. I guess the fact that it did the wrong thing was just conveniently hidden because we didn't remove the containers before.